hub/api/dataset.py

```diff
@@ -152,9 +121,9 @@ def __getitem__(
                 return self.tensors[item][self.index]
         elif isinstance(item, (int, slice, list, tuple, Index)):
             return Dataset(
-                read_only=self.read_only,
```

Comment:
we don't need to pass read)only anymore, will automatically be inferred from storage.